### PR TITLE
Routing: Update alerting template routes

### PR DESCRIPTION
**Pull request by Clarity-89**
**What is this feature?**

Update alerting template routes to react router 6. These are the child routes of the routes defined in `alerting/routes` file, so they are do not need to be hoisted there. 

**Why do we need this feature?**

Part of the react router migration. 

**Which issue(s) does this PR fix?**:

Part of https://github.com/grafana/grafana/issues/92065

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/features/alerting/unified/Templates.tsx [hunk 1]
```diff
@@ -1,31 +1,23 @@
...
 import { AlertmanagerPageWrapper } from './components/AlertingPageWrapper';
-
-const EditMessageTemplate = SafeDynamicImport(() => import('./components/contact-points/EditMessageTemplate'));
-const NewMessageTemplate = SafeDynamicImport(() => import('./components/contact-points/NewMessageTemplate'));
-const DuplicateMessageTemplate = SafeDynamicImport(
-  () => import('./components/contact-points/DuplicateMessageTemplate')
-);
+import DuplicateMessageTemplate from './components/contact-points/DuplicateMessageTemplate';
+import EditMessageTemplate from './components/contact-points/EditMessageTemplate';
+import NewMessageTemplate from './components/contact-points/NewMessageTemplate';
```
Comment (Clarity-89) [inline public/app/features/alerting/unified/Templates.tsx:8]: This component is already lazy loaded in the [routes file](https://github.com/grafana/grafana/blob/main/public/app/features/alerting/routes.tsx#L111-L119), so it's not necessary to do so here, also it's harder to do with the new router API.

File: public/app/features/alerting/unified/Templates.test.tsx [hunk 2]
```diff
@@ -38,6 +39,15 @@ jest.mock('@grafana/ui', () => ({
...
 }));
 
+const successMock = jest.fn();
+jest.mock('app/core/copy/appNotification', () => {
+  return {
+    useAppNotification: () => {
+      return { success: successMock };
+    },
+  };
+});
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/Templates.test.tsx:49]: (Going to add a commit to remove the mock, as we can still render `AppNotificationList` in the setup method and assert it that way :+1:)

File: public/app/features/alerting/unified/Templates.tsx [hunk 3]
```diff
@@ -1,31 +1,23 @@
...
-        exact={true}
-        path="/alerting/notifications/templates/:name/duplicate"
-        component={DuplicateMessageTemplate}
-      />
-    </Switch>
+    <Routes>
+      <Route path=":name/edit" element={<EditMessageTemplate />} />
+      <Route path="new" element={<NewMessageTemplate />} />
+      <Route path=":name/duplicate" element={<DuplicateMessageTemplate />} />
+    </Routes>
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/Templates.tsx:20]: Do we ideally want to hoist these to the main alerting routes file as well? I know you mentioned in the description that it's not required, but what is the end goal/pattern for managing routes? Or is it up to us (within the scope of Alerting)?

Comment (Clarity-89) [reply public/app/features/alerting/unified/Templates.tsx:20]: These are proper child routes, as opposed to the duplicate routes in other places, so I think they can stay here. The main issue was with the routes defined separately and in the routes file, making their migration harder.

**Review by tomratcliffe — APPROVED** [review]
Thank you! 🚀

**Pull request merged**